cmd/limactl/edit.go

```diff
@@ -51,12 +54,33 @@ func editAction(cmd *cobra.Command, args []string) error {
 		return errors.New("Cannot edit a running instance")
 	}
 
+	flags := cmd.Flags()
+
+	if runtime.GOOS != "windows" && flags.Lookup("start-at-login").Changed {
+		startAtLogin, err := flags.GetBool("start-at-login")
```

Comment:
How is this to be used?

```console
$ l edit --start-at-login=true
INFO[0000] The autostart file ("/Users/jan/Library/LaunchAgents/io.lima-vm.autostart.default.plist") has been created
Waiting for Emacs...
INFO[0004] Aborting, no changes made to the instance
```

So it asks me to edit `lima.yaml`, and if I don't make some unrelated change, it aborts the operation.

Compare with

```console
$ l edit --cpus 3
INFO[0000] Instance "default" configuration edited
? Do you want to start the instance now?  (Y/n)
FATA[0004] interrupt
```

Somehow the `--start-at-login` option doesn't really fit in with the `edit` subcommand because it doesn't change any `lima.yaml` settings.

Also `edit` doesn't work on running instance, which is not a sensible limitation for enabling/disabling autostart:

```console
$ l edit --start-at-login
FATA[0000] Cannot edit a running instance
```

So maybe this should be a separate subcommand instead of being shoehorned into `edit`?

Reply:
So the `--start-at-login` flag might be present in the `lima.yaml`? How do you see this? I'm a bit confused with the usage flow.

Reply:
Not sure; needs more discussion and feedback from other @lima-vm/maintainers.

Adding a `startAtLogin` setting to `lima.yaml` would fix part of the problem, but still wouldn't allow you to change the setting without stopping the instance first, which feels like an unnatural limitation.

`start-at-login` also doesn't feel like it is a configuration setting for the instance: it is not used at all for starting or configuring the instance; it is a configuration for the host on which the instance exists. Maybe that is too subtle a difference, but I think that is the reason why the semantics are different from other `lima.yaml` settings, and why it may not belong in there.

E.g. I can create a clone of the default instance with

```console
$ l create --name clone ~/.lima/default/lima.yaml
```

Should this copy the `--start-at-login` setting from the default instance too? I would think not, which is another sign that maybe the setting doesn't belong into `lima.yaml`.

Sorry, I don't have a constructive suggestion on what we should do; I need to think about it some more. But I would love to hear other people's opinions!

Reply:
This is just a suggestion for discussion; let's get consensus before you make any code changes:

A new command will create/delete the start-at-login files:

```
limactl start-at-login [INSTANCE] [--enabled]
```

This will create the file unless `--enabled=false` is specified, in which case it will be deleted. `INSTANCE` defaults to `default`, as usual.

The command can be executed as long as `INSTANCE` exists; it doesn't matter what state it is in.

`limactl create ... --start-at-login` could still be supported as a convenience, but it is unnecessary complexity[^1] for extremely minor benefit. I would prefer to not have it (I would want to keep it for compatibility if we had already released it, but we didn't).

[^1]: I'm talking about the user experience complexity, not the implementation complexity. The more options each command gets, the more overwhelming it becomes for new users to figure out what is important, and what isn't.

`limactl delete` will still delete the unit file, as it does right now.

What do you think?

Reply:
@jandubois 

Thank you for your thoughtful suggestion and for initiating this discussion.

Introducing a new command to manage `start-at-login` files could offer clarity and consistency in our workflow. Your proposed command structure seems intuitive and aligns well with our existing conventions.

The current implementation seems to have a drift with `lima.yaml` config file. In my view, the `create\start\edit` works primarily with the config file, and an additional flag should be included in `lima.yaml` in order to follow the internal flow.